### ISSUE TITLE
[BUG] documents with "owners" disappear after selecting date-range

### ISSUE THREAD

**Issue by Skram-de** [paperless-ngx/paperless-ngx#3203]
### Description

Documents with an owner set disappear in document list after selecting a date range (see below) for "added" or "created".
Selecting a specific date works properly.
Documents without an owner are shown properly also.

### Steps to reproduce

1. login as a superuser
2. go to web-ui -> documents
3. reset filters (all documents displayed now)
4. select "created" oder "added" date range "last ...." 
5. do not use specific date "before" or "after"
6. documents with owner set disappear now....

### Webserver logs

```bash
nothing here
```

### Paperless-ngx version

1.14.0

### Host OS

Linux 3.10.108 #42661 SMP (x86_64)

### Installation method

Docker - official image

### Browser

Chrome

**Comment by shamoon** [comment]
<img width="1372" alt="Screenshot 2023-04-27 at 12 35 39 AM" src="https://user-images.githubusercontent.com/4887959/234792539-9ee5d242-5577-449b-b0c2-4bb33873e229.png">

I cant reproduce this. Perhaps theres something more specific?

**Comment by Skram-de** [comment]
Please use "Last year" to reproduce. Using "after date" already works fine.

**Comment by shamoon** [comment]
That would be an odd bug, still not seeing (first doc has owner)

<img width="1378" alt="Screenshot 2023-04-27 at 1 36 32 AM" src="https://user-images.githubusercontent.com/4887959/234807594-02692d7d-c4d5-4b91-b2f8-6ed7ffdb21a0.png">

**Comment by shamoon** [comment]
Try reindexing, `python3 manage.py document_index reindex`

**Comment by Skram-de** [comment]
Reindexing didn't fix it.
It seems, selecting an "last-date-field" would break the search.
Example: I selected one storage path and a **specific** date. Works as excepted:
![image](https://user-images.githubusercontent.com/113503248/234828497-6e83938a-fe38-44fc-ae02-7f77734c9f3a.png)

Now I selected "Last 7 days" and the storage path is ignored:
![image](https://user-images.githubusercontent.com/113503248/234829141-da129d00-5871-4649-a8e4-31dede057f06.png)
